client/state/partner-portal/products/selectors.ts

```diff
@@ -0,0 +1,46 @@
+import { PartnerPortalStore, APIProductFamilyProduct } from 'calypso/state/partner-portal/types';
+// Required for modular state.
+import 'calypso/state/partner-portal/init';
+
+export function getSelectedProductSlugs( state: PartnerPortalStore ) {
+	return state.partnerPortal.products.selectedProductSlugs;
+}
+
+export function getDisabledProductSlugs(
+	state: PartnerPortalStore,
+	allProducts: APIProductFamilyProduct[]
+) {
+	const selectedProductSlugs = getSelectedProductSlugs( state );
+
+	return (
+		selectedProductSlugs
+			// Get the product objects corresponding to the selected product slugs
+			.map( ( selectedProductSlug ) =>
+				allProducts?.find( ( product ) => product.slug === selectedProductSlug )
+			)
+			// Get all the product slugs of products within the same product family as the selected product
+			.flatMap( ( selectedProduct ) =>
+				allProducts
+					?.filter(
+						( product ) =>
+							product.family_slug === selectedProduct?.family_slug &&
+							product.slug !== selectedProduct.slug
+					)
+					.map( ( product ) => product.slug )
+			)
+	);
+}
+
+export function getTotalSelectedCost(
+	state: PartnerPortalStore,
+	allProducts: APIProductFamilyProduct[]
+) {
+	const selectedProductSlugs = getSelectedProductSlugs( state );
+
+	const totalSelectedProductCost = allProducts
+		.filter( ( product ) => selectedProductSlugs.includes( product.slug ) )
+		.map( ( product ) => product.amount )
+		.reduce( ( totalCost, productCost ) => totalCost + productCost, 0 );
+
+	return totalSelectedProductCost;
```

Comment:
I think we can get rid of 2 iterations here by doing: 

```suggestion
	return allProducts.reduce(
		( totalCost, product ) =>
			// Include only selected products for calculating total cost
			totalCost + ( selectedProductSlugs.includes( product.slug ) ? product.amount : 0 ),
		0
	);
```

Reply:
For this case I prefer to leave it as is. The suggested change does improve the calculation from O(3*n) to O(1*n), but the overall time complexity is still O(n) in both cases and n is quite small. In such cases it is best to prioritize having easily comprehendable code which is broken up into simple statements.